tests/test_freqtradebot.py

```diff
@@ -3585,34 +3582,43 @@ def test_get_real_amount_no_currency_in_fee(default_conf, trades_for_order, buy_
     )
     freqtrade = get_patched_freqtradebot(mocker, default_conf)
 
-    # Amount does not change
-    assert freqtrade.get_real_amount(trade, limit_buy_order) == amount
+    if not use_ticker_rate:
+        mocker.patch('freqtrade.exchange.Exchange.fetch_ticker', side_effect=ExchangeError)
 
+    caplog.clear()
+    # Amount does not change
+    assert freqtrade.get_real_amount(trade, buy_order) == amount - fee_reduction_amount
+
+    if expected_log:
+        assert log_has(expected_log, caplog)
+
+
+@pytest.mark.parametrize(
+    'stake_currency, fee_cost, fee_currency, fee_reduction_amount, expected_fee, expected_log', [
+        ("BTC", None, None, 0.001, 0.001, (
+            'Applying fee on amount for Trade(id=None, pair=LTC/ETH, amount=8.00000000, '
+            'open_rate=0.24544100, open_since=closed) (from 8.0 to 7.992).'
+        )),
+        ("ETH", 0.02, 'BNB', 0.0005, 0.001518575, (
+            'Applying fee on amount for Trade(id=None, pair=LTC/ETH, amount=8.00000000, '
+            'open_rate=0.24544100, open_since=closed) (from 8.0 to 7.996).'
```

Comment:
It looks like the 2 strings in these test are identical, so just make the number at the end a parameter